### ISSUE TITLE
Get avatar from user

### ISSUE THREAD

**Issue by jianghoy** [eamodio/vscode-gitlens#1036]
There's already some feature request out there asking for avatar from [Github](https://github.com/eamodio/vscode-gitlens/issues/281) or [Gitlab](302). However, some users, like me, work in a corporate environment, so these avatar sources do not work well in this case. Besides, these requests have not been fulfilled so far.

It would be nice to allow users to choose their avatars from local source.

**Comment by jianghoy** [comment]
I'm willing to commit a PR if we nail down the specs.

**Comment by sgtwilko** [comment]
Simplest solution could be to point to a folder containing images with the email address, these could even be stored in the repo.

**Comment by michaelblyons** [comment]
I would also like such a thing to exist. I thought I had asked for it in the past, but apparently not. Then I can map my coworkers to their official photos, rather than whatever icon they have (sometimes none) on Gravatar.

Even now that the GitHub images are coming, I'd rather use my own images.

----

> Simplest solution could be to point to a folder containing images with the email address.

I do not like this as much as a map. If "Jane" has several email addresses she has used, I don't want to make an image for each one. Just one `jane.jpg` and be done.

>  These could even be stored in the repo.

We have lots of repos. I don't want to put duplicate images in all of them. And really I do not want to check in the images at all.

**Comment by eamodio** [comment]
Maybe there can just be a setting that is a map of e-mails to file uris?

**Comment by michaelblyons** [comment]
> Maybe there can just be a setting that is a map of e-mails to file uris?

Yes. That's exactly what I imagined. (All your avatar lookups happen after `.mailmap` transforms, right?)

**Comment by sgtwilko** [comment]
> 
> > Simplest solution could be to point to a folder containing images with the email address.
> 
> I do not like this as much as a map. If "Jane" has several email addresses she has used, I don't want to make an image for each one. Just one `jane.jpg` and be done.

It would seem to me that a map to point to a folder is just doubling the work.
You have to manage the folder images and the mapping.

If the image size is an issue (for avatar?) the user could use symbolic links to point to the same image.

If a co-worker has multiple emails you'd have to do some work duplicating the values in the map or duplicating the image.

How many co-workers actually have multiple emails in your repos?

All our repos required an email address from our AzureAD.

> 
> >  These could even be stored in the repo.
> 
> We have lots of repos. I don't want to put duplicate images in all of them. And really I do not want to check in the images at all.

Fairnuff, didn't say the feature had to point to the repo, just saying it _could_ be stored there.

I'd assume that with or without a map you'd have to have a setting to specify the folder location?

**Comment by eamodio** [comment]
I'm honestly fine with either approach:

1. New setting which is a hash (map) of `email`: `filename`
1. New setting which points to a folder which would contain images that match the email address

Or a 3rd approach of a combination of both:

3. New setting that could be either -- a `string` path, or a hash

**Comment by michaelblyons** [comment]
Is there actually missing info or just a tag that is still sitting around? I feel like Eric's last comment pretty much covers any "need more info" stuff.

**Comment by eamodio** [comment]
@michaelblyons just a bad label, sorry for the noise.

**Comment by Goddard** [comment]
What gource does, is they use the username .jpg or .png...etc.  This is inside the .git/avatars/ directory.  I think this is what a lot of projects are doing.  Some people make scripts to pull from gravatar, but you don't have to do that.  You could just as easily place whatever image you want in that folder.

So if I worked on the project it would be "Ryein Goddard.png" and this can be from .mailmap or not.

**Comment by atbeta** [comment]
Really hope to add this feature.

**Comment by jsun969** [comment]
This feature would be very helpful. #1636 would be a good implementation for integrations as well.
